cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -330,15 +331,39 @@ Status TransferInt(RecordReader* reader, MemoryPool* pool,
   auto values = reinterpret_cast<const ParquetCType*>(reader->values());
   auto out_ptr = reinterpret_cast<ArrowCType*>(data->mutable_data());
   std::copy(values, values + length, out_ptr);
+  int64_t null_count = 0;
+  std::vector<std::shared_ptr<Buffer>> buffers = {nullptr, std::move(data)};
   if (field->nullable()) {
-    *out = std::make_shared<ArrayType<ArrowType>>(field->type(), length, std::move(data),
-                                                  reader->ReleaseIsValid(),
-                                                  reader->null_count());
-  } else {
-    *out =
-        std::make_shared<ArrayType<ArrowType>>(field->type(), length, std::move(data),
-                                               /*null_bitmap=*/nullptr, /*null_count=*/0);
+    null_count = reader->null_count();
+    buffers[0] = reader->ReleaseIsValid();
+  }
+  auto array_data =
+      ::arrow::ArrayData::Make(field->type(), length, std::move(buffers), null_count);
+  auto array_statistics = std::make_shared<::arrow::ArrayStatistics>();
+  array_statistics->null_count = null_count;
+  auto statistics = metadata->statistics().get();
+  if (statistics) {
+    if (statistics->HasDistinctCount()) {
+      array_statistics->distinct_count = statistics->distinct_count();
+    }
+    if (statistics->HasMinMax()) {
+      auto typed_statistics =
+          static_cast<::parquet::TypedStatistics<ParquetType>*>(statistics);
+      const ArrowCType min = typed_statistics->min();
+      const ArrowCType max = typed_statistics->max();
+      if (std::is_signed<ArrowCType>::value) {
+        array_statistics->min = static_cast<int64_t>(min);
+        array_statistics->max = static_cast<int64_t>(max);
+      } else {
+        array_statistics->min = static_cast<uint64_t>(min);
+        array_statistics->max = static_cast<uint64_t>(max);
+      }
+      array_statistics->is_min_exact = true;
+      array_statistics->is_max_exact = true;
```

Comment:
Can you add correspond comment here? This might be a bit tricky

Reply:
Sure. We should document about the discussion at https://github.com/apache/arrow/pull/43595#discussion_r1706467430 , right?

BTW, could you share the e-mail URL for https://github.com/apache/arrow/pull/43595#discussion_r1719243469 ?

> I guess no, I'll send a mail to maillist to make it sure

I couldn't find it at https://lists.apache.org/list.html?dev@parquet.apache.org .

Ah, I forgot to add a writer check here. I should have set `true` only when a writer is Apache Parquet C++. I'll fix it.

Reply:
Sorry, let me setup a issue, generally if it's from Parquet C++, it will works. I'm a bit busy this morning preparing for my tour, I'll try to work it out this noon

Reply:
No problem. Thanks!

Reply:
https://lists.apache.org/thread/zfnmg5p51b7oylft5w5k4670wgkd4zv4

Reply:
Thanks!

Reply:
I've added a comment that we can always use `true` for integer based min/max.

I didn't need `if (::arrow::internal::StartsWith(ctx->reader->metadata()->created_by(), "parquet-cpp-arrow"))` for this case based on your e-mail.

Reply:
FYI, I found the string and FLBA might being truncated, other types in public impl will not being truncated if exists